### PR TITLE
Add mTLS logging for authentication policy prototyping

### PR DESCRIPTION
**Pull request by lei-tang**
**What this PR does / why we need it**: For authentication policy prototyping, add mTLS logging.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: src/envoy/http/authn/mtls_authentication.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Http {
+
+class MtlsAuthentication : public Logger::Loggable<Logger::Id::http> {
```
Comment (qiwzhang) [inline src/envoy/http/authn/mtls_authentication.h]: please add comment per c++ style guide

https://google.github.io/styleguide/cppguide.html

Comment (lei-tang) [reply src/envoy/http/authn/mtls_authentication.h]: Comments have been added.

**Pull request merged**